### ISSUE TITLE
Module build failed: Duplicate declaration "indexes" in combination with webpack as bundler

### ISSUE THREAD

**Issue by bdering** [Automattic/mongoose#6547]
**Do you want to request a *feature* or report a *bug*?**
I would like to report a bug, but this bug occurs in combination with **webpack** only. It seems to be a declaration of a **global** variable named **"indexes"** somewhere else in the code. I'm using node js version **v8.9.0**.

**What is the current behavior?**
The current behavior is that, during the bundle time of webpack in node js mode I receiving following error: 

`ERROR in ./node_modules/mongoose/lib/services/schema/indexes.js
Module build failed: Duplicate declaration "indexes"

  10 |
  11 | module.exports = function indexes(schema) {
> 12 |   let indexes = [];
     |       ^
  13 |   const schemaStack = [];
  14 |   const indexTypes = schema.constructor.indexTypes;
  15 |

 @ ./node_modules/mongoose/lib/schema.js 17:14-50
 @ ./node_modules/mongoose/lib/index.js
 @ ./node_modules/mongoose/index.js`

**If the current behavior is a bug, please provide the steps to reproduce.**
Steps to reproduce:

1. install dependencies: 
`$ yarn add mongoose 
$ yarn add webpack webpack-cli uglifyjs-webpack-plugin --dev`
2.  import mongoose anywhere in app.js like here:
`import mongoose from 'mongoose';
` in this case you should also install babel latest sytax and create a .babelrc
3. use this webpack config:

```
let path = require('path');
let webpack = require('webpack');
const BUILD_PATH = path.resolve(__dirname, './dist');
const UglifyJSPlugin = require('uglifyjs-webpack-plugin');

module.exports = {
  entry: ['babel-polyfill', './app/index.js'],
  output: {
    path: BUILD_PATH,
    filename: 'bundle.js'
  },
  module: {
    rules: [
      {
        test: /\.js$/,
        exclude: '/node_modules/',
        loader: 'babel-loader'
      }
    ]
  },
  plugins: [
    new webpack.HotModuleReplacementPlugin({
      multiStep: true
    }),
    new webpack.NoEmitOnErrorsPlugin(),
    new webpack.DefinePlugin({
      'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV)
    })
  ],
  optimization: {
    minimizer: [
      new UglifyJSPlugin({
        uglifyOptions: {
          warning: "verbose",
          ecma: 7,
          beautify: false,
          compress: false,
          comments: false,
          mangle: false,
          toplevel: false,
          keep_classnames: true,
          keep_fnames: true
        }
      })
    ]
  },
  target: 'node',
  node: {
    __filename: false,
    __dirname: false
  },
  devtool: 'eval-source-map',
};
```

**What is the expected behavior?**
No build (bundle) time error. By the way, with version 4.0 and above this bug disappears. 

**Please mention your node.js, mongoose and MongoDB version.**
Nodejs **v8.9.0**

**Comment by Pixelatex** [comment]
Can confirm, same issue here.

**Comment by vkarpov15** [comment]
Found the issue, it looks like this is the same problem as [this babel issue](https://github.com/babel/minify/issues/611#issuecomment-314388652).

```javascript
function indexes(schema) {
  // Duplicate because it has the same name as the function
  let indexes = [];
}
```

I'd argue this is incorrect behavior on babel's side, but since this function name isn't externally facing we fixed it by just renaming `indexes()` -> `getIndexes()`, webpack build will succeed with 5.1.5

**Comment by vkarpov15** [comment]
Fix is in 3ac9f5ba048b866304dba944cd2d18e7d2b25f4c, I just made a typo in the commit message #6457 -> 6547.

**Comment by vkarpov15** [comment]
And another typo, 2e5ac5a4ab9b09d69b11c4c170172068277667b8 should refer to this issue not #6457
